presto-main/src/main/java/com/facebook/presto/operator/scalar/DataSizeFunctions.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.LiteralParameters;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlType;
+import io.airlift.slice.Slice;
+import io.airlift.units.DataSize;
+
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+import static com.facebook.presto.spi.type.StandardTypes.DOUBLE;
+import static io.airlift.units.DataSize.Unit.BYTE;
+
+public class DataSizeFunctions
+{
+    private DataSizeFunctions() {}
+
+    @Description("converts data size string to bytes")
+    @ScalarFunction("parse_data_size")
+    @LiteralParameters("x")
+    @SqlType(DOUBLE)
+    public static double parseDataSize(@SqlType("varchar(x)") Slice dataSize)
```

Comment:
In `parse_duration` we return number of milliseconds as `long`. I wonder this function should also return long. But when returning long there is a high chance of petabyte values overflow. The value should be higher than `9,223,372PB`, so i'm not sure if the overflow is something real.

Reply:
This method doesn't support parsing zetta bytes. I'm not sure how likely that someone will be parsing something `9000000PB`.

> Maybe this should return decimal(38)

I was thinking about it. It won't increase precision. The `DataSize` class uses `double`.